debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlErrorHandler.java

```diff
@@ -39,6 +41,9 @@ else if (throwable instanceof ServerException) {
         else if (throwable instanceof DebeziumException && throwable.getCause() != null) {
             return isRetriable(throwable.getCause());
         }
+        else if (throwable instanceof SocketException || throwable instanceof EOFException) {
```

Comment:
Could we perhaps further narrow down that condition, like a status code or (parts of) the error message?

Reply:
I tracked the exception stack of the code, BinaryLogClient does not throw a wrapped exception but io EOFExcetion or network SocketException. So... any other good ideas?